### DISCUSSION TITLE
Add pop-up warning when adding Scene on Tileset  if Scene uses said resource

### DISCUSSION THREAD

**Discussion by SanNull** [godotengine/godot-proposals#8990] [Editor]
Adding scenes on the Tileset resource is handy, especially when it is reusable, however it can cause unforeseen issues.  E.g: you have a tile.res which both _SceneA_ and **SceneB** use. You would like to instanciate the latter into _SceneA_  by using the tileset. By adding **SceneB** into the resource, it creates a paradox (adding **SceneB** into the same resource it is used to instanciate it), and either the tiles for _A_ or **B** gets deleted. I suggest adding a warning whenever this situation occurs, in order to make Godot feel more reliable and less error-prone.

**Comment by AThousandShips** [comment]
This sounds like a general circular reference limitation, which happens in many other places, and it's not easy to discover at times

**Reply by SanNull** [reply]
Agreed. I still believe there should be a way to prevent this so less of it happens.
